src/rule-creator.ts

```diff
@@ -3,15 +3,20 @@ import { ESLintUtils } from '@typescript-eslint/experimental-utils'
 import type { NGRX_MODULE } from './utils'
 import { docsUrl, ngrxVersionSatisfies, NGRX_MODULE_PATHS } from './utils'
 
-export type NgRxRuleModule<TMessageIds extends string = string> =
-  TSESLint.RuleModule<TMessageIds, unknown[]> & {
-    meta: { ngrxModule: NGRX_MODULE; version?: string }
-  }
-
-type CreateRuleMetaDocs = Omit<TSESLint.RuleMetaDataDocs, 'url'>
+type Meta<TMessageIds extends string> =
+  | TSESLint.RuleMetaData<TMessageIds> & {
+      ngrxModule?: NGRX_MODULE
+      version?: string
+    }
```

Comment:
Ideally we should do this instead (I tried this but TS doesn't feel happy with this):

```suggestion
type Meta<TMessageIds extends string> =
  | TSESLint.RuleMetaData<TMessageIds>
  | (TSESLint.RuleMetaData<TMessageIds> & {
      ngrxModule: NGRX_MODULE
      version: string
    })
```

In other words, it either should **not** contain any of the extra properties (`ngrxModule` and `version`) or it **should** contain both properties. Currently, we can pass only one of them 😞.